### PR TITLE
*: s/time.Unix/timeutil.Unix/ to ensure UTC

### LINKED ISSUE #18774: cli: regression in test Example_sql_format
I simply ran `make test PKG=./pkg/cli` on a recent master on my Linux system - no other changes.
This seems to have regressed with #18676.

```
--- FAIL: Example_sql_format (0.15s)
got:
sql -e create database t; create table t.times (bare timestamp, withtz timestamptz)
CREATE TABLE
sql -e insert into t.times values ('2016-01-25 10:10:10', '2016-01-25 10:10:10-05:00')
INSERT 1
sql -e select * from t.times
bare    withtz
2016-01-25 05:10:10+00:00       2016-01-25 15:10:10+00:00
# 1 row
want:
sql -e create database t; create table t.times (bare timestamp, withtz timestamptz)
CREATE TABLE
sql -e insert into t.times values ('2016-01-25 10:10:10', '2016-01-25 10:10:10-05:00')
INSERT 1
sql -e select * from t.times
bare    withtz
2016-01-25 10:10:10+00:00       2016-01-25 15:10:10+00:00
# 1 row
```

### PR DESCRIPTION
**Pull request by dt**
timeutil.Unix wraps stdlib's time.Unix but ensures the result is in UTC, not Local.
time.Unix() uses the time.Local zone, but that is host-dependant.

Since we expect geographically diverse clusters of interchangable nodes, host-local
time zones are not desireable.

Fixes #18774.

### CODE REVIEW

**Comment by knz** [comment]
Huh, I don't understand this. Please extend the commit message with a discussion of what's going on here.

**Comment by knz** [comment]
LGTM.

cc @cockroachdb/build @cockroachdb/core

**Comment by dt** [comment]
After thinking about this a bit more, I became worried there were probably other paths by which we'd let a `time.Local`'ed `time.Time` make its way into a datum, so I just replaced it with a much more compressive change to replace `time.Unix` with a wrapper to ensure it always gets `UTC`ed.

File: pkg/sql/parser/datum.go [hunk 1]
```diff
@@ -38,10 +38,15 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
 	"github.com/cockroachdb/cockroach/pkg/util/ipaddr"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/uint128"
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 )
 
+func init() {
+	time.Local = time.UTC
```
Comment (tamird) [inline pkg/sql/parser/datum.go]: um

File: pkg/util/timeutil/time.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ var UnixEpoch = time.Unix(0, 0)
 // time, usec microseconds since UnixEpoch. In Go's current time.Time
```
Comment (tamird) [inline pkg/util/timeutil/time.go]: this comment is no longer correct.

File: pkg/util/timeutil/now_windows.go [hunk 3]
```diff
@@ -41,5 +42,5 @@ func init() {
 func Now() time.Time {
```
Comment (tamird) [inline pkg/util/timeutil/now_windows.go]: this comment still talks about local time.

File: pkg/util/timeutil/now_unix.go [hunk 4]
```diff
@@ -22,5 +22,5 @@ import (
 
 // Now returns the current local time.
```
Comment (tamird) [inline pkg/util/timeutil/now_unix.go]: this comment still talks about local time.

**Comment by knz** [comment]
Reviewed 37 of 38 files at r1, 10 of 10 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[pkg/sql/parser/datum.go, line 47 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-KuzMetuMCbnsvcTeFvC-r2-47:-KuzPWf1WJl-y61vlnI7:b-lmy66l) ([raw file](https://github.com/cockroachdb/cockroach/blob/9f3fa797ea2d9cd01774b0df2d8c1a1b6118ce0d/pkg/sql/parser/datum.go#L47)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

um
</blockquote></details>

Yeah now you've introduced timeutil this is not warranted any more.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783)*

File: pkg/sql/parser/datum.go (hunk 1 above)
Comment (dt) [reply pkg/sql/parser/datum.go]: heh, oh, yeah, that was what this commit was supposed to avoid.

File: pkg/util/timeutil/now_unix.go (hunk 4 above)
Comment (dt) [reply pkg/util/timeutil/now_unix.go]: Done.

File: pkg/util/timeutil/now_windows.go (hunk 3 above)
Comment (dt) [reply pkg/util/timeutil/now_windows.go]: Done.

File: pkg/util/timeutil/time.go (hunk 2 above)
Comment (dt) [reply pkg/util/timeutil/time.go]: Done.

**Comment by dt** [comment]
Okay, various import cycles and comments fixed up now. RFAL!

**Comment by knz** [comment]
LGTM with questions

---

Reviewed 10 of 18 files at r3, 10 of 10 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[pkg/rpc/context_test.go, line 528 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-KuzyXabuI6Orbj9sVFC:-KuzyXacyj-9CI4iGvMP:b-dl8zch) ([raw file](https://github.com/cockroachdb/cockroach/blob/3095785e9a78dd1d37a22f6f37e56d1fd696f8aa/pkg/rpc/context_test.go#L528)):*
> ```Go
> 	ac.time = time.Add(ac.getAdvancementInterval())
> 	ac.Unlock()
> 	return time.UnixNano()
> ```

Why the revert?

---

*[pkg/util/timeutil/now_windows.go, line 44 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-Kuzyl5FIw4EquiM3ZRG:-Kuzyl5FIw4EquiM3ZRH:bg7tlnx) ([raw file](https://github.com/cockroachdb/cockroach/blob/3095785e9a78dd1d37a22f6f37e56d1fd696f8aa/pkg/util/timeutil/now_windows.go#L44)):*
> ```Go
> 	var ft windows.Filetime
> 	windows.GetSystemTimePreciseAsFileTime(&ft)
> 	return time.Unix(0, ft.Nanoseconds()).UTC()
> ```

Why not use timeutil here too?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-:-KuzynSN5dYLw3nvUnQA:b-bnc1mu)*

**Comment by dt** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[pkg/rpc/context_test.go, line 528 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-KuzyXabuI6Orbj9sVFC:-KuzzFM8TtqmAu0V9HFN:b7s4l6m) ([raw file](https://github.com/cockroachdb/cockroach/blob/3095785e9a78dd1d37a22f6f37e56d1fd696f8aa/pkg/rpc/context_test.go#L528)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Why the revert?
</blockquote></details>

`time` there is a local var named time, not the package `time`.

---

*[pkg/util/timeutil/now_windows.go, line 44 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783#-Kuzyl5FIw4EquiM3ZRG:-KuzzMdmdM8nJ6Kqiab5:b-c5t0iu) ([raw file](https://github.com/cockroachdb/cockroach/blob/3095785e9a78dd1d37a22f6f37e56d1fd696f8aa/pkg/util/timeutil/now_windows.go#L44)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Why not use timeutil here too?
</blockquote></details>

we're *in* `timeutil` here -- we could just use `Unix()` but eh: this file is one of the few exempted from the timeutil linter and thus assumed to be responsible doing the right thing on its own.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18783)*

**Comment by knz** [comment]
Aw, thanks.

**Comment by dt** [comment]
@knz hmm, this feels like a bit big for a cherry-pick at this point... thoughts?

**Comment by knz** [comment]
It's spread out but the change is systematic. I do not see a problem. If in doubt you can wait for 1.1.1 but it should make it to 1.1 eventually (together with the other ts fix in #18764)

**Pull request merged**